src/test/java/com/company/bitmanipulation/SingleNumberTwoTest.java

```diff
@@ -0,0 +1,25 @@
+package com.company.bitmanipulation;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class SingleNumberTwoTest {
+    @ParameterizedTest
+    @MethodSource("testDataProvider")
+    void testSingleNumber(int[] arr, int expectedResult) {
```

Comment:
access modifier. Not just here. review it everywhere in code

Reply:
sure